service/controller/app/resource/chart/desired.go

```diff
@@ -68,26 +70,26 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
+	repositories := []string{repositoryURL}
```

Comment:
@kubasobon you have written this part I believe, so not sure if you are ok with this change, to me it feels bit more readable now, but also changes one thing in the logic. Previously it has been falling back to other repositories only on the `notFoundError` seen with the primary one. But then for the fallback repositories the type of error has not been checked. So this to me feels like we do not really need to check for this in the primary case as well, but maybe you had a specific reason for doing this that way.

Reply:
Looks good. I think it's ok to round-robin through repositories on other kinds of errors as well. I was overly cautious with initial design, I think.